prow/spyglass/lenses/junit/lens.go

```diff
@@ -136,48 +193,65 @@ func (lens Lens) Body(artifacts []lenses.Artifact, resourceDir string, data stri
 	}
 	sort.Slice(results, func(i, j int) bool { return results[i].path < results[j].path })
 
-	jvd := struct {
-		NumTests int
-		Passed   []TestResult
-		Failed   []TestResult
-		Skipped  []TestResult
-	}{}
+	var jvd JVD
+
 	for _, result := range results {
 		if result.err != nil {
 			continue
 		}
-		for _, test := range result.junit {
-			if test.Failure != nil {
+		for _, tests := range result.junit {
```

Comment:
The logic for figuring out flakiness seems complicated enough to get its own function (which should make testing it a bit easier too?)

Reply:
would prefer not, transmitting these 4 states between 2 functions seems a bit too much to me